pop/service/api.go

```diff
@@ -0,0 +1,215 @@
+package service
+
+import (
+	"bytes"
+
+	"github.com/BurntSushi/toml"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/base64"
+	"github.com/satori/go.uuid"
+	"gopkg.in/dedis/onet.v1"
+	"gopkg.in/dedis/onet.v1/log"
+	"gopkg.in/dedis/onet.v1/network"
+)
+
+const (
+	// ErrorWrongPIN indicates a wrong PIN
+	ErrorWrongPIN = iota + 4100
+	// ErrorInternal indicates something internally went wrong - see the
+	// error message
+	ErrorInternal
+	// ErrorOtherFinals indicates that one or more of the other conodes
+	// are still missing the finalization-step
+	ErrorOtherFinals
+)
+
+func init() {
+	network.RegisterMessage(&FinalStatement{})
+	network.RegisterMessage(&PopDesc{})
+}
+
+// FinalStatement is the final configuration holding all data necessary
+// for a verifier.
+type FinalStatement struct {
+	// Desc is the description of the pop-party.
+	Desc *PopDesc
+	// Attendees holds a slice of all public keys of the attendees.
+	Attendees []abstract.Point
+	// Signature is created by all conodes responsible for that pop-party
+	Signature []byte
+}
+
+// The toml-structure for (un)marshaling with toml
+type finalStatementToml struct {
+	Desc      *popDescToml
+	Attendees []string
+	Signature string
+}
+
+// NewFinalStatementFromString creates a final statement from a string.
+func NewFinalStatementFromString(s string) *FinalStatement {
```

Comment:
why string ? In toml there is `toml.Unmarshal` which takes a slice of bytes. I'm just wondering if there is a reason or not. The caller of the library, will always have to do the casting `string(myBytesRead)` because he'll read `[]bytes` most of the time with `ioutil.ReadFile(statementPath)`, so maybe change that to `[]byte` ?